archivist_samples/wipp/run.py

```diff
@@ -0,0 +1,282 @@
+            ),
+        ],
+    )
+    LOGGER.info("Tomography registered...")
+
+    # Loading
+    LOGGER.info("Loading drum and cask...")
+    drum.loading(
+        {
+            "description": "Loaded drum into "
+            + cask.asset["attributes"]["arc_display_name"],
+            "container": cask.asset["identity"],
+        },
+        attachments=[
+            upload_attachment(arch, "trupact_loading.jpg", "arc_primary_image")
+        ],
+    )
+    cask.loading(
+        {
+            "description": "Filled with "
+            + drum.asset["attributes"]["arc_display_name"],
+            "container": cask.asset["identity"],
+        },
+        custom_asset_attrs={
+            "wipp_inventory": drum.asset["identity"],
+        },
+        attachments=[
+            upload_attachment(arch, "trupact_loading.jpg", "arc_primary_image")
+        ],
+    )
+    LOGGER.info("Loading registered...")
+
+    # Pre-shipping
+    LOGGER.info("Pre-shipping inspection...")
+    drum.preshipping(
+        {
+            "description": "Inspection inventory "
+            + cask.asset["attributes"]["arc_display_name"],
+        },
+        attachments=[
+            upload_attachment(arch, "preshipment_inspection.jpg", "arc_primary_image")
+        ],
+    )
+    cask.preshipping(
+        {
+            "description": "Inspected " + cask.asset["attributes"]["arc_display_name"],
+        },
+        attachments=[
+            upload_attachment(arch, "preshipment_inspection.jpg", "arc_primary_image")
+        ],
+    )
+    LOGGER.info("Pre-shipping inspection registered...")
+
+    # Departure
+    LOGGER.info("Loading departure...")
+    drum.departure(
+        {
+            "description": "Departed SRS inventory "
+            + cask.asset["attributes"]["arc_display_name"],
+        },
+        attachments=[
+            upload_attachment(arch, "truck_departure.jpg", "arc_primary_image"),
+            upload_attachment(
+                arch, "SRS_to_WPP_route_instructions.pdf", "approved_route"
+            ),
+        ],
+    )
+    cask.departure(
+        {
+            "description": cask.asset["attributes"]["arc_display_name"]
+            + "departing for WIPP."
+        },
+        attachments=[
+            upload_attachment(arch, "truck_departure.jpg", "arc_primary_image"),
+            upload_attachment(
+                arch, "SRS_to_WPP_route_instructions.pdf", "approved_route"
+            ),
+        ],
+    )
+    LOGGER.info("Departure registered...")
+
+    # Waypoint
+    LOGGER.info("Loading waypoints...")
+    cask.waypoint(
+        {
+            "description": "TRAGIS smart sensors ping: All sensors GREEN",
+        },
+        custom_attrs={
+            "wipp_sensors_shock": "0",
+            "wipp_sensors_rad": "45",
+        },
+        attachments=[
+            upload_attachment(arch, "truck_departure.jpg", "arc_primary_image")
+        ],
+    )
+    LOGGER.info("Waypoint registered...")
```

Comment:
This needs to actually register waypoint data. That means:
* The event needs to actually have a GIS location on it
* There need to be lots of them! The route is specified in the attachment to the first transit run

So you need a loop that records these waypoints:

    waypoints = [["Atlanta", "33.592177", "-84.406064"],
                 ["Talladega", "33.592177", "-86.248379"],
                 ["Birmingham", "33.494993", "-86.895403"],
                 ["Tuscaloosa", "33.184220", "-87.610330"],
                 ["Meridian", "32.391672", "-88.532850"],
                 ["Jackson", "32.285409", "-90.074633"],
                 ["Monroe", "32.463868", "-91.893769"],
                 ["Shreveport", "32.537993", "-93.651582"],
                 ["Tyler", "32.334001", "-95.321504"],
                 ["South Dallas", "32.639816", "-96.826631"],
                 ["Gordon", "32.499115", "-98.521317"],
                 ["Abilene", "32.457004", "-99.816598"],
                 ["Big Spring", "32.244259", "-101.458984"],
                 ["Andrews", "32.312469", "-102.548197"],
                 ["Seminole", "32.457004", "-99.816598"],
                 ["Hobbs", "32.244259", "-101.458984"]]

Reply:
Checked a new run, looks good!